_posts/2024-04-09-flake-schemas.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: post
+title:  "Flake Schemas"
+date:   2024-04-09 20:00:00
+categories: nix, flakes, schemas, experimental
+github_comments_issueid: "30"
+---
+
+## Introduction
+
+Flake schemas were [introduced by Determinate Systems](https://determinate.systems/posts/flake-schemas/) (written by [Eelco Dolstra](https://github.com/edolstra) --- creator of Nix) about half a year ago now (August 2023), which is quite a promising feature, but what is the current status? Is it usable yet?
+
+The [PR](https://github.com/NixOS/nix/pull/8892) that introduces flake schemas was submitted to `NixOS/nix` the same day the announcement was made, and it sadly remains in DRAFT mode until today. However, if you look at the comments, some expert users reported using this feature successfully, so I thought it was about time to give it a shot myself.
+
+So here's where the fun starts: in order to use it, we need *that* specific `nix` client version (from the PR) with flake schemas support, which fortunately for us, it's conveniently provided on [this flake](https://github.com/DeterminateSystems/nix-src/tree/flake-schemas).
+
+## Sneak preview
+
+Before we look into the shenanigans we need to get through to get this working, let's appreciate for a second what it brings to the table. 
+
+Shown below is the `nix flake show` output of my [NixOS configuration](https://github.com/gvolpe/nix-config) repo:
+
+![schemas](../images/flake-schemas.png)
+
+Neat! Don't you think? I was irked by the usual `homeConfigurations: unknown` output.
```

Comment:
**suggestion:** before showing the "improved" output, show what the same flake looks like without flake-schemas first? better demo effect, and people reading this a couple years in the future may not even know what `homeConfigurations: unknown` was (in the happy path scenario)

Reply:
yeah that's a good point, I'll add a screenshot on how it looks nowadays :)